### PR TITLE
Pgcraft cleanups

### PR DESCRIPTION
**Pull request by mbouron**

### CODE REVIEW

File: libnopegl/src/pipeline_compat.h [hunk 1]
```diff
@@ -27,7 +27,10 @@
 #include "pipeline.h"
 
 struct pipeline_compat_params {
-    const struct pipeline_params *params;
+    int type;
```
Comment (ubitux) [inline libnopegl/src/pipeline_compat.h:30]: can you add `NGLI_PIPELINE_TYPE_*` in comment?

File: libnopegl/src/pgcraft.c [hunk 2]
```diff
@@ -236,13 +236,8 @@ static const char *get_glsl_type(int type)
...
 {
     int *next_bind = s->next_bindings[type];
-    if (!next_bind) {
-        /*
-         * Non-explicit bindings is still allowed for OpenGL context not
-         * supporting explicit locations/bindings.
-         */
-        return -1;
-    }
+    ngli_assert(next_bind);
```
Comment (ubitux) [inline libnopegl/src/pgcraft.c:260]: Typo `pgraft`/`pgcraft` in commit desciption

File: libnopegl/src/pgcraft.c [hunk 3]
```diff
@@ -235,7 +253,10 @@ static const char *get_glsl_type(int type)
 
 static int request_next_binding(struct pgcraft *s, int type)
 {
-    int *next_bind = s->next_bindings[type];
+    ngli_assert(type >= 0 && type <= NGLI_TYPE_NB);
```
Comment (ubitux) [inline libnopegl/src/pgcraft.c:256]: `< NGLI_TYPE_NB`

**Review by ubitux — CHANGES_REQUESTED** [review]

File: libnopegl/src/pgcraft.c (hunk 3 above)
Comment (mbouron) [reply libnopegl/src/pgcraft.c]: Fixed.

File: libnopegl/src/pgcraft.c (hunk 2 above)
Comment (mbouron) [reply libnopegl/src/pgcraft.c:260]: Fixed.

File: libnopegl/src/pipeline_compat.h (hunk 1 above)
Comment (mbouron) [reply libnopegl/src/pipeline_compat.h]: Done.

**Review by ubitux — APPROVED** [review]

**Pull request merged**